spec.html

```diff
@@ -435,7 +435,7 @@ <h1>Evaluate ( ) Concrete Method</h1>
           1. Assert: _m_.[[Status]] is `"evaluating"`.
           1. Set _m_.[[Status]] to `"evaluated"`.
           1. <del>Set _m_.[[EvaluationError]] to _result_.</del>
-          1. <ins>Perform ! CyclicModuleExecutionRejected(_m_, _result_).</ins>
+          1. <ins>If _m_ is not the last element of _stack_, perform ! CyclicModuleExecutionRejected(_m_, _result_).</ins>
```

Comment:
I think we need another line here as well to remove the last element of stack to ensure this handles the full stack?

Reply:
Hmm actually I'm not quite sure I follow this fix I must admit, it seems like it will just reject the last element of stack everytime then?

Reply:
Sorry, I don't understand. I don't think this stack will have duplicates in it. What should we remove?

Reply:
The last element of the stack is the one that threw the exception, so it will already be rejected. That's why we're skipping it. I think when we're down and just executing the synchronous module, it's not the right place to skip the CyclicModuleExecutionRejected call, as there may or may not be async parents at that point (and if there are async parents, we will probably not end up doing this stack walk).

Reply:
I think we actually have another but here and that is that the synchronous executions that run during the initial InnerModuleEvaluation are supposed to be throw completions of InnerModuleEvaluation which isn't happening either unfortunately.

Reply:
That's why I keep suggesting (and originally did) just make these their own execution calls. (so as a result, the last element of stack won't be the one that threw an exception!)

There will never be async parents for a sync module that has no async child modules in its tree (ie exactly those sync modules that execute during InnerModuleEvaluate, and can throw from it).

Reply:
For these reasons, I do think https://github.com/tc39/proposal-top-level-await/issues/84#issuecomment-496997675 might be the easiest fix.

Reply:
What makes you say that these completions aren't being thrown? Isn't this what would happen in step 14.b of InnerModuleEvaluation?

Reply:
For https://github.com/tc39/proposal-top-level-await/issues/84#issuecomment-496997675 , I still don't understand who would trigger the rejection actions for a synchronous module that depends on something async and has other modules depending on it.